library/private/options_tools.h.in

```diff
@@ -348,6 +348,47 @@ std::string getAsString(const options& opt, const std::string& name)
     throw options::no_value_exception("Trying to get " + name + " before it was set");
   }
 }
+
+//----------------------------------------------------------------------------
+/**
+ * Generated method, see `options::isOptional`
+ */
+bool isOptional(const std::string& name)
+{
+  // clang-format off
+  ${_options_is_optional};
+  // clang-format on
+  else throw options::inexistent_exception("Option " + name + " does not exist");
+}
+
+//----------------------------------------------------------------------------
+/**
+ * Generated method, see `options::isOptional`
+ */
+void reset(options& opt, const std::string& name)
+{
+  // clang-format off
+  ${_options_reset};
+  // clang-format on
+  else throw options::inexistent_exception("Option " + name + " does not exist");
+}
+
+//----------------------------------------------------------------------------
+/**
+ * Generated method, see `options::removeValue`
+ */
+void removeValue(options& opt, const std::string& name)
+{
+  if (isOptional(name))
+  {
+    reset(opt, name);
+  }
+  else
+  {
+    throw options::incompatible_exception("Option " + name + " is not not optional");
+  }
+}
```

Comment:
you can put that in options.cxx directly

Reply:
done